### PR TITLE
Case imports can now be done asynchronously

### PR DESCRIPTION
**Pull request by archendel**
Signed-off-by: LE SAULNIER Kevin <kevin.lesaulnier@rte-france.com>

**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [X] The commit message follows our guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the current behavior?** *(You can also link to an open issue here)*
Case can be imported synchronously, from a rest API endpoint

**What is the new behavior (if this is a feature change)?**
Case can be imported asynchronously as well, it's now the default behaviour if the user does not specify a param

### CODE REVIEW

File: src/main/java/com/powsybl/network/conversion/server/NotificationService.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package com.powsybl.network.conversion.server;
```
Comment (FranckLecuyer) [inline src/main/java/com/powsybl/network/conversion/server/NotificationService.java:1]: Missing license and author

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java [hunk 2]
```diff
@@ -131,6 +138,33 @@ private static EquipmentInfos toEquipmentInfos(Identifiable<?> i, UUID networkUu
...
+
+            String caseFormat = getCaseFormat(caseUuid);
+            try {
+                networkInfos = importCase(caseUuid, variantId, reportUuid, importParameters);
+            } catch (Exception e) {
+                LOGGER.error(e.getMessage(), e);
+                notificationService.emitCaseImportFailed(receiver, e.getMessage());
+                return;
+            }
+            notificationService.emitCaseImportSucceeded(networkInfos, caseFormat, receiver);
```
Comment (FranckLecuyer) [inline src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java:164]: could be moved inside the try ?

File: src/test/java/com/powsybl/network/conversion/server/NetworkConversionTest.java [hunk 3]
```diff
@@ -117,25 +131,26 @@ public void test() throws Exception {
...
-            MvcResult mvcResult = mvc.perform(post("/v1/networks")
+            mvc.perform(post("/v1/networks")
                 .param("caseUuid", UUID.randomUUID().toString())
-                .param("reportUuid", UUID.randomUUID().toString()))
+                .param("reportUuid", UUID.randomUUID().toString())
+                .param("isAsyncRun", "false"))
                 .andExpect(status().isOk())
                 .andReturn();
 
-            assertEquals("{\"networkUuid\":\"" + randomUuid + "\",\"networkId\":\"20140116_0830_2D4_UX1_pst\"}",
```
Comment (FranckLecuyer) [inline src/test/java/com/powsybl/network/conversion/server/NetworkConversionTest.java:126]: why removing the assertion ? import is done synchronously here, like before ...

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/main/java/com/powsybl/network/conversion/server/NotificationService.java (hunk 1 above)
Comment (archendel) [reply src/main/java/com/powsybl/network/conversion/server/NotificationService.java:1]: fixed

File: src/test/java/com/powsybl/network/conversion/server/NetworkConversionTest.java (hunk 3 above)
Comment (archendel) [reply src/test/java/com/powsybl/network/conversion/server/NetworkConversionTest.java:126]: mixed things up while I was developping
fixing

**Review by FranckLecuyer — APPROVED** [review]

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java [hunk 4]
```diff
@@ -94,16 +97,20 @@ public class NetworkConversionService {
...
+    private final NotificationService notificationService;
+
     private final ObjectMapper objectMapper;
 
     @Autowired
     public NetworkConversionService(@Value("${backing-services.case-server.base-uri:http://case-server/}") String caseServerBaseUri,
                                     @Value("${backing-services.geo-data-server.base-uri:http://geo-data-server/}") String geoDataServerBaseUri,
                                     @Value("${backing-services.report-server.base-uri:http://report-server}") String reportServerURI,
-                                    NetworkStoreService networkStoreService, @Lazy EquipmentInfosService equipmentInfosService, NetworkConversionExecutionService networkConversionExecutionService) {
+                                    NetworkStoreService networkStoreService, @Lazy EquipmentInfosService equipmentInfosService, NetworkConversionExecutionService networkConversionExecutionService,
```
Comment (laurgarn) [inline src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java:108]: Could we take the chance to insert a newline before `@Lazy` and between the other non terminal comma and the next N ?

**Review by laurgarn — COMMENTED** [review]
Read, nothing spotted.

**Review by AbdelHedhili — APPROVED** [review]

File: src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java (hunk 4 above)
Comment (archendel) [reply src/main/java/com/powsybl/network/conversion/server/NetworkConversionService.java:108]: fixed

**Pull request merged**